inc/Engine/Media/Lazyload/Content/Processor/Dom.php

```diff
@@ -0,0 +1,68 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Media\Lazyload\Content\Processor;
+
+use DOMDocument;
+
+class Dom {
+	public function add_locations_hash_to_html( $html ) {
+		// Load HTML into DOMDocument.
+		$dom = new DOMDocument();
+
+		@$dom->loadHTML( $html, LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD );
```

Comment:
Maybe it would be good to do proper error handling instead of the @ syntax? We could maybe add a log with the logger in case an error occurs, and bail out?